commonknowledge/mapped
# feat: display pledged groups on hub map in separate layer

## Description
Splits out group pledges into a separate data source, so they can be displayed
separately on the map.

## Motivation and Context
https://linear.app/commonknowledge/issue/CC-197/display-group-pledge-sign-ups-on-the-map

## How Can It Be Tested?
1. Create a new AirTable for the group pledges. Make sure it has the following columns:
- EMAIL
- POSTCODE
- FNAME
- LNAME
- GROUP_NAME
- GROUP_URL
- GROUP_SOCIAL
- HEARD_FROM
- MAP_CONSENT
2. Create a new data source for this AirTable. Set up the mappings:
- POSTCODE => postcode
- GROUP_NAME => title
- GROUP_URL => publicUrl
- GROUP_SOCIAL => socialUrl
- MAP_CONSENT => can display point
- EMAIL => email

## How Will This Be Deployed?
Standard deployment, then a new data source will need to be set up for Climate Coalition. 

In the database, this data source will need `can_display_details_publicly` and `can_display_points_publicly` set to true:

```
UPDATE hub_externaldatasource SET can_display_details_publicly=True, can_display_points_publicly=True
  WHERE id='the-new-data-source-id'
```

and added to the Hub Data config in wagtail:

```
  {
    "id": "the-new-data-source-id",
    "name": "Groups",
    "type": "groups",
    "source": "the-new-data-source-id",
    "geom_field": "point",
    "icon_image": "tcc-canvassing"
  }
```

Finally, the Member Form block in puck will need its settings updated. The Group Data Source needs to be set to the new data source that has just been created above ^.
